atlasdb-workload-server/src/test/java/com/palantir/atlasdb/workload/store/AtlasDbTransactionStoreTest.java

```diff
@@ -0,0 +1,158 @@
+
+    private static final WorkloadCell WORKLOAD_CELL_ONE =
+            ImmutableWorkloadCell.builder().key(50).column(10).build();
+    private static final WorkloadCell WORKLOAD_CELL_TWO =
+            ImmutableWorkloadCell.builder().key(1257).column(521).build();
+    private static final WorkloadCell WORKLOAD_CELL_THREE =
+            ImmutableWorkloadCell.builder().key(567).column(405234).build();
+
+    private static final Integer VALUE_ONE = 100;
+
+    private TransactionManager manager;
+
+    private AtlasDbTransactionStore store;
+
+    @Before
+    public void before() {
+        manager = TransactionManagers.createInMemory(Set.of());
+        store = AtlasDbTransactionStore.create(
+                manager,
+                Map.of(
+                        TABLE_REFERENCE,
+                        AtlasDbUtils.tableMetadata(ConflictHandler.SERIALIZABLE),
+                        INDEX_REFERENCE,
+                        AtlasDbUtils.indexMetadata()));
+    }
+
+    @Test
+    public void createsTables() {
+        assertThat(manager.getKeyValueService().getAllTableNames()).contains(TABLE_REFERENCE, INDEX_REFERENCE);
+    }
+
+    @Test
+    public void canWriteDataToStore() {
+        Optional<WitnessedTransaction> witnessedTransactionPrimaryTable =
+                store.readWrite(List.of(WriteTransactionAction.of(TABLE, WORKLOAD_CELL_ONE, VALUE_ONE)));
+        Optional<WitnessedTransaction> witnessedTransactionIndexTable =
+                store.readWrite(List.of(WriteTransactionAction.of(INDEX_TABLE, WORKLOAD_CELL_TWO, VALUE_ONE)));
+        assertThat(witnessedTransactionPrimaryTable).isPresent();
+        assertThat(witnessedTransactionIndexTable).isPresent();
+        assertThat(store.get(TABLE, WORKLOAD_CELL_ONE)).contains(VALUE_ONE);
+        assertThat(store.get(INDEX_TABLE, WORKLOAD_CELL_TWO)).contains(VALUE_ONE);
+    }
+
+    @Test
+    public void witnessedTransactionMaintainsOrder() {
+        List<WitnessedTransactionAction> actions = List.of(
+                WitnessedWriteTransactionAction.of(TABLE, WORKLOAD_CELL_TWO, 100),
+                WitnessedReadTransactionAction.of(TABLE, WORKLOAD_CELL_TWO, Optional.of(100)),
+                WitnessedReadTransactionAction.of(TABLE, WORKLOAD_CELL_THREE, Optional.empty()),
+                WitnessedWriteTransactionAction.of(TABLE, WORKLOAD_CELL_ONE, 24),
+                WitnessedReadTransactionAction.of(TABLE, WORKLOAD_CELL_ONE, Optional.of(24)));
+        Optional<WitnessedTransaction> maybeTransaction = store.readWrite(actions.stream()
+                .map(action -> action.accept(WitnessToActionVisitor.INSTANCE))
+                .collect(Collectors.toList()));
+        assertThat(maybeTransaction).isPresent();
+        WitnessedTransaction transaction = maybeTransaction.get();
+        assertThat(transaction.commitTimestamp()).isNotEmpty();
+        assertThat(transaction.actions()).containsExactlyElementsOf(actions);
```

Comment:
Generally, for this you can do `assertThat(maybeTransaction).hasValueSatisfying(value -> { ... })`

Reply:
ok yeah that's a lot cleaner!